### ISSUE TITLE
[Feature request] Kanban View

### ISSUE THREAD

**Issue by djpeinado** [johannesjo/super-productivity#374]
Hello,
First of all thank you for your great application, and thanks for open sourcing.

I'd really appreciate also a Kanban view for the tasks. I didn't find any open source offline tasks manager application with a Kanban view, and I think many people are also looking for this.
I don't know if it could be very difficult, somehow incompatible or out of the main scope, but I wanted to suggest.

Thanks in advance

**Comment by johannesjo** [comment]
Hey! Thank you very much for opening this up! I'd like this feature to be added at some point as well. Currently there is a lot of other stuff on my plate for this project, so this might take a while. Meanwhile Pull Requests and more detailed suggestions for this are welcome.

**Comment by johannesjo** [comment]
Still relevant.

**Comment by dalekirkwood** [comment]
I would also love to see this, something like the ToDoIst Kanban. Very easy to set up in the GUI and customizable.

**Comment by dalekirkwood** [comment]
@johannesjo Am I missing something :-) ?

**Comment by johannesjo** [comment]
@dalekirkwood no. By the emoji I just ment that I will have a look if I ever find the time to work on this.:)

**Comment by Shahin-rmz** [comment]
Hello,
thanks for the good app.
Although I do not use super productivity in my workflow but I really adore it.
Is there any API (perforably for Python) so I can sync my self hosted [kanban](https://kanboard.org/) with super productivity?
Thanks
Another alternative is to implement kanban in GUI of the application, which is because of lacking of time is not possible at moment.

**Comment by johannesjo** [comment]
> Is there any API (perforable for Python) so I can sync my self hosted [kanban](https://kanboard.org/) with super productivity?

@Shahin-rmz no, there is not :)

**Comment by AlanSanchezP** [comment]
I came here just to upvote this request, and to add another possible alternative for those who can find it suitable for their needs:

[Pomodoro Logger](https://github.com/zxch3n/PomodoroLogger) has an integrated kanban board per project. I couldn't find a way to use the pomodoro timer in more than one list per board (it has a "focused list" marker that tells the app which list represents your ongoing work and will display the tasks in that list in the timer view)

**Comment by S7venLights** [comment]
Pitty it's only built for desktop, I need time tracking on the move

**Comment by S7venLights** [comment]
I have started a list of kanban apps with time trackers and I'll add there if I find any.
https://github.com/stars/S7venLights/lists/time-tracking-with-kanban

**Comment by dalekirkwood** [comment]
I switched to TickTick. It doesnt really have time tracking but its an
amazing app.

**Comment by S7venLights** [comment]
Right now I don't really use Superprodictivity to add new tasks for every day and mark them as completed. I make a task for general types of things I do everyday, eg: Getting ready in the morning, Personal Emails, TV, etc and I track everything. Then I can monitor where I'm wasting time or what work I did for what client. This also makes the app faster, as there aren't completed tasks piling up… 
If there was a kanban board I might change how I use the app, but there would have to be a way for completed tasks to auto delete after a while to keep things fast

**Comment by S7venLights** [comment]
Tick Tick does look nice based on the screenshots, I like that it has a calendar, but it's full of trackers and permissions

**Comment by Cyber-Syntax** [comment]
Please, consider to focalboard features too. I didn't mention their markdown editor, it's useless but, their property feature must be in that view. Also, you can consider later to add table view...

If you are considering adding this view instead of the focalboard integration, I also recommend making project-based note-taking pages.

For example, you can look at the notion app for ideas. Of course, it is not for you to do the same, but I think an application where we can take notes, see and track our projects in a kanban view would be great.

**Comment by Cyber-Syntax** [comment]
still relevant

**Comment by gelatinbomb** [comment]
Still Relevant!

**Comment by cobbinaloic** [comment]
Still relevant

**Comment by StillNewIdeas** [comment]
Howdy! 
 I'm currently busy finishing my apprenticeship as an Software Engineer with a final project and I absolutely adore superproductivity. 

I always had a problem with time management, so your app really established a game-changer in my workflow.
Absolutely love it. If there would be a slightest chance to get a kanban-view as well, it would be perfect. The biggest issue I'm facing right now is that my ToDo's for the day can be quiet overwhelming and kanban is just a great method to manage tasks in a better overview without using too many tags. Combined with the 'Focus Mode' (my all-time-fav) it would be just so good to have it.

I'm getting more and more to the point where I want to bury myself into the code and figure it out myself, because it would be a really cool enhancement and I know that you have so much to work on. 

Thanks again and especially thank you for this astonishing piece of work!
Bless you.

**Comment by Jagdfalke** [comment]
> Howdy! I'm currently busy finishing my apprenticeship as an Software Engineer with a final project and I absolutely adore superproductivity.
> 
> I always had a problem with time management, so your app really established a game-changer in my workflow. Absolutely love it. If there would be a slightest chance to get a kanban-view as well, it would be perfect. The biggest issue I'm facing right now is that my ToDo's for the day can be quiet overwhelming and kanban is just a great method to manage tasks in a better overview without using too many tags. That is my current workflow, that I have 3 Tags, representing the classical kanban-columns and I add them to my tags. Biggest downside is the fact that you have to remove one to avoid having one task in multiple columns. So a board-view, coombined with the 'Focus Mode' (my all-time-fav) it would be just so good to have it.
> 
> I'm getting more and more to the point where I want to bury myself into the code and figure it out myself, because it would be a really cool enhancement and I know that you have so much to work on.
> 
> Thanks again and especially thank you for this astonishing piece of work! Bless you.

Hi, Thank you very much for your feedback!

Seeing you are a software engineer we would really appreciate if you considered looking into developing the feature yourself! 

I know that @johannesjo is all for this feature. It is however a significant addition to Super Producitivity and it will take some time to implement.

**Comment by cobbinaloic** [comment]
Yes, still relevant

**Comment by Cyber-Syntax** [comment]
I'm really surprised that this super useful feature has flown under the radar for so long. 

I think what's making it hard is that it's integrated with scheduling and other stuff. If we could just isolate this feature on the dashboard and give us some basic controls - like labels(do,doing,done) and drag-and-drop functionality - I think we could use it effectively while still working on improving it.

**Comment by cobbinaloic** [comment]
I agree with you.

**Comment by C3rberus99** [comment]
Yes, still relevant

**Comment by johannesjo** [comment]
> Maybe UI wise their could be a kanban button. So you can easily turn a project or list of tasks you are viewing like Today tasks into a kanban board.

I would likely put it to the left hand menu below schedule & planner.

**Comment by dalekirkwood** [comment]
Here after leaving a comment 2 years ago. Still paying for TickTick

**Comment by storchg** [comment]
I would cry tears of joy, where a good thing to happen. Please, we need this.

**Comment by S7venLights** [comment]
Seems SuperProductivity integrates with [Open Project](https://www.openproject.org/) which seems to have a kanban

**Comment by storchg** [comment]
Really cool, but I am tied to W*nd*ws and iiuc open project doesn't allow integration of repositories when running in docker

**Comment by Designz48** [comment]
What programming language is needed to code for this app? I've been using Javascript to code my own Kanban's in Obsidian.

Also absolutely love this todo app! It's everything, and I mean EVERYTHING I've needed after moving from Notion. Flexible, easy to manage, while not being too much customization. Ya'll are incredible for adding all the needed features!

**Comment by dalekirkwood** [comment]
Just for anyone looking. I started to use Vikunja. It's open source and has both Kanban & Gaant charts.

**Comment by johannesjo** [comment]
This will be added with this next release! Closing this!

**Comment by KatrinIhler** [comment]
> This will be added with this next release! Closing this!

Just noticed this and the Eisenhower matrix in the new update when I came back from vacation - best start into the work week I could've hoped for ❤ Thank you for all the work you do!

**Comment by storchg** [comment]
Thanks, Johannes! Made my Week!

**Comment by S7venLights** [comment]
> [Vikunja](https://try.vikunja.io/)

It looks like a great project but it doesn't do time tracking
